sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlan.scala

```diff
@@ -355,7 +357,21 @@ abstract class SparkPlan extends QueryPlan[SparkPlan] with Logging with Serializ
 
   protected def newPredicate(
       expression: Expression, inputSchema: Seq[Attribute]): GenPredicate = {
-    GeneratePredicate.generate(expression, inputSchema)
+    try {
+      GeneratePredicate.generate(expression, inputSchema)
```

Comment:
It is good to control it using an option. This is not a part of the whole-stage codegen.
Is it better to use `sqlContext.conf.wholeStageFallback` or add `sqlContext.conf.codegenFallback`?
What do you think?